Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAuditdPlugin.py

```diff
@@ -49,94 +68,299 @@ def Get_Marshall(Ensure):
+    return os.path.join(GetAgentConfDir(WorkspaceId), "auditd_plugin.conf")
+
+def GetTmpAuomsOutputConfPath(WorkspaceId):
+    return os.path.join(GetTmpDir(WorkspaceId), "auoms.conf")
+
+def GetAuomOutputConfPath(WorkspaceId):
+    return os.path.join(AUOMS_OUTCONF_DIR, WorkspaceId + ".conf")
+
+def GetTmpAuditRulesPath(WorkspaceId):
+    return os.path.join(GetTmpDir(WorkspaceId), "auditd_plugin.rules")
+
+def Set(WorkspaceId, Ensure):
+    if not IsValidWorkspaceId(WorkspaceId):
+        return [1]
+
+    (audit_version, plugin_state,
+     actual_audit_rules, actual_outconf, actual_plugin_conf) = GetState(WorkspaceId)
+    if audit_version is None:
+        return [1]
+
+    (desired_plugin_state, desired_rules, desired_outconf,
+     desired_plugin_conf) = GetDesiredState(WorkspaceId, Ensure, audit_version, plugin_state)
+    if desired_plugin_state is None:
+        return [1]
+
+    arg1 = ""
+    if IsTextDifferent(actual_plugin_conf, desired_plugin_conf):
+        LG().Log(LOG_INFO, "Actual Plugin conf differs from Desired plugin conf so omsagent will be restarted")
+        arg1 = "restart"
+        if desired_plugin_conf is None:
+            RemoveFile(GetOMSAgentConfPath(WorkspaceId))
+        else:
+            WriteFile(GetOMSAgentConfPath(WorkspaceId), desired_plugin_conf)
+
+    rules_file_path = ""
+    if IsTextDifferent(actual_audit_rules, desired_rules):
+        if desired_rules is None:
+            rules_file_path = "remove"
+        else:
+            rules_file_path = GetTmpAuditRulesPath(WorkspaceId)
+            WriteFile(rules_file_path, desired_rules)
+
+    outconf_name = ""
+    outconf_path = ""
+    if IsTextDifferent(actual_outconf, desired_outconf):
+        if not desired_outconf is None:
+            outconf_name = WorkspaceId + ".conf"
+            outconf_path = GetTmpAuomsOutputConfPath(WorkspaceId)
+            WriteFile(outconf_path, desired_outconf)
 
     # Run script that will restart omsagent and enable+start/disable+stop auoms
     args = []
     args.append("/usr/bin/sudo")
     args.append("/opt/microsoft/omsconfig/Scripts/OMSAuditdPlugin.sh")
-    if Ensure == "Present":
-        args.append("enable")
+    args.append("set")
+    args.append(arg1)
+    if plugin_state != desired_plugin_state:
+        args.append(desired_plugin_state)
     else:
-        args.append("disable")
+        args.append("")
+    args.append(rules_file_path)
+    args.append(outconf_name)
+    args.append(outconf_path)
     proc = subprocess.Popen(args, -1, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
     exit_code = proc.wait()
     if exit_code != 0:
-        out_txt = proc.stdout.read().decode('utf8','replace').encode('utf8','replace')
-        err_txt = proc.stderr.read().decode('utf8','replace').encode('utf8','replace')
-        LG().Log('INFO', "stdout: " + out_txt)
-        LG().Log('INFO', "stderr: " + err_txt)
+        out_txt = proc.stdout.read().decode('utf8', 'replace').encode('utf8', 'replace')
```

Comment:
GitHub is saying these two lines are different from the ones above; did you change spacing?